core/logger/logger.go

```diff
@@ -163,16 +205,132 @@ func (l *zapLogger) withCallerSkip(skip int) Logger {
+		scope.SetContext("logger", map[string]interface{}{
+			"args": args,
+		})
+		scope.SetLevel(sentry.LevelFatal)
+	})
+	hub.CaptureMessage(fmt.Sprintf("%v", args))
+	l.withCallerSkip(1).Fatal(args...)
+}
+
+func (l *zapLogger) Panic(args ...interface{}) {
+	defer sentry.Flush(SentryFlushDeadline)
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", map[string]interface{}{
+			"args": args,
+		})
+		scope.SetLevel(sentry.LevelFatal)
+	})
+	hub.CaptureMessage(fmt.Sprintf("%v", args))
+	l.withCallerSkip(1).Panic(args...)
+}
+
+func (l *zapLogger) Errorf(format string, values ...interface{}) {
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", map[string]interface{}{
+			"values": values,
+		})
+		scope.SetLevel(sentry.LevelError)
+	})
+	hub.CaptureMessage(fmt.Sprintf(format, values...))
+	l.withCallerSkip(1).Errorf(format, values...)
+}
+
+func (l *zapLogger) Fatalf(format string, values ...interface{}) {
+	defer sentry.Flush(SentryFlushDeadline)
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", map[string]interface{}{
+			"values": values,
+		})
+		scope.SetLevel(sentry.LevelFatal)
+	})
+	hub.CaptureMessage(fmt.Sprintf(format, values...))
+	l.withCallerSkip(1).Fatalf(format, values...)
+}
+
+func (l *zapLogger) Errorw(msg string, keysAndValues ...interface{}) {
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", toMap(keysAndValues))
+		scope.SetLevel(sentry.LevelError)
+	})
+	hub.CaptureMessage(msg)
+	l.withCallerSkip(1).Errorw(msg, keysAndValues...)
+}
+
+func (l *zapLogger) Fatalw(msg string, keysAndValues ...interface{}) {
+	defer sentry.Flush(SentryFlushDeadline)
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", toMap(keysAndValues))
+		scope.SetLevel(sentry.LevelFatal)
+	})
+	hub.CaptureMessage(msg)
+	l.withCallerSkip(1).Fatalw(msg, keysAndValues...)
+}
+
+func (l *zapLogger) Panicw(msg string, keysAndValues ...interface{}) {
+	defer sentry.Flush(SentryFlushDeadline)
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", toMap(keysAndValues))
+		scope.SetLevel(sentry.LevelFatal)
+	})
+	hub.CaptureMessage(msg)
+	l.withCallerSkip(1).Panicw(msg, keysAndValues...)
+}
+
+func toMap(args ...interface{}) (m map[string]interface{}) {
+	m = make(map[string]interface{}, len(args)/2)
+	for i := 0; i < len(args); {
+		// Make sure this element isn't a dangling key
+		if i == len(args)-1 {
+			break
+		}
+
+		// Consume this value and the next, treating them as a key-value pair. If the
+		// key isn't a string ignore it
+		key, val := args[i], args[i+1]
+		if keyStr, ok := key.(string); ok {
+			m[keyStr] = val
+		}
+		i += 2
+	}
```

Comment:
Is this simplification valid?
```suggestion
	for i := 0; i+1 < len(args); i+=2 {
		// Consume this value and the next, treating them as a key-value pair. If the
		// key isn't a string ignore it
		key, val := args[i], args[i+1]
		if keyStr, ok := key.(string); ok {
			m[keyStr] = val
		}
	}
```